compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -65,6 +77,50 @@ func (n *NodeInventoryCollector) Scan(nodeName string) (*storage.NodeInventory,
 	return m, nil
 }
 
+// callAnalyzer isolates the call to Analyze to ensure we recover from any panic that might occur.
+// Analyze runs on a potentially unknown live file system, so it might be prone to unexpected behavior.
```

Comment:
Add something like: We isolate this function to protect the component against the analyzerFunction : (1) panicking, (2) taking too long to execute (and potentially consuming node resources).